### PR TITLE
Add com.github.emilianavt.OpenSeeFace (CLI app)

### PR DESCRIPTION
**Pull request by orowith2os**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/emilianavt/OpenSeeFace/issues/43#issuecomment-1191453978
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by orowith2os** [comment]
Marked as draft, because there still needs to be all of the pip dependencies added.

File: com.github.emilianavt.OpenSeeFace.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+id: com.github.emilianavt.OpenSeeFace
+branch: master
+runtime: org.freedesktop.Platform
+runtime-version: "21.08"
+sdk: org.freedesktop.Sdk
+command: 'facetracker.sh'
+finish-args:
+    - "--device=all"
+    - "--share=network"
```
Comment (hfiguiere) [inline com.github.emilianavt.OpenSeeFace.yml:9]: no display permissions?

does it really need network access?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.github.emilianavt.OpenSeeFace.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+id: com.github.emilianavt.OpenSeeFace
+branch: master
```
Comment (hfiguiere) [inline com.github.emilianavt.OpenSeeFace.yml:2]:
```suggestion
```

**Comment by hfiguiere** [comment]
also missing: .desktop file, appstream file, icons (128x128 or SVG)

File: com.github.emilianavt.OpenSeeFace.yml (hunk 1 above)
Comment (orowith2os) [reply com.github.emilianavt.OpenSeeFace.yml:9]: This app is meant to be used as a CLI app, so no display permissions needed.
It also requires network access to output to a port on the network for the face tracking data.

**Comment by orowith2os** [comment]
I don't believe any of those are needed for this CLI app, but I can add the appstream file and icons if required.

**Comment by hfiguiere** [comment]
Appstream is required.

**Comment by orowith2os** [comment]
I've added the metainfo.xml. Now it needs to install the pip dependencies, and, for some reason, the command for it )using python /app/bin/facetracker.py) is broken.

File: com.github.emilianavt.OpenSeeFace.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,113 @@
+<?xml version='1.0' encoding='utf-8'?>
+<component type="desktop">
```
Comment (hfiguiere) [inline com.github.emilianavt.OpenSeeFace.metainfo.xml:2]:
```suggestion
<component type="console-application">
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
> I've added the metainfo.xml. Now it needs to install the pip dependencies, and, for some reason, the command for it )using python /app/bin/facetracker.py) is broken.

https://github.com/flatpak/flatpak-builder-tools should help. There is a pip and poetry version.

**Comment by orowith2os** [comment]
Using the flatpak tools to bundle the pip dependencies didn't work, specifically with onnxruntime. Instead, I plan on utilizing a method that the latest alpha release of vpuppr uses, where it uses pyinstaller to make a facetracker binary that HOPEFULLY should work. I'll look into this method for the madness in a bit.

**Comment by orowith2os** [comment]
That didn't seem to work. It turns out there's an issue with the pip flatpak builder utility, so I've submitted an issue there.

**Pull request closed without merging**